### DISCUSSION TITLE
Is it possible to set triggers on the storage schema?

### DISCUSSION THREAD

**Discussion by shobhitsrivastava** [supabase/supabase#18021] [Questions]
Hi, I've been trying to set a new trigger on the storage schema, directly through the sql editor. I confirmed the function call is correct, but getting "Invalid SQL query" when I try to create the trigger, without any other info. Does supabase allow creating triggers on storage.objects table?

```
CREATE TRIGGER create_transcription_job_trigger
AFTER INSERT ON storage.objects 
FOR EACH ROW
EXECUTE FUNCTION create_new_transcription_job(NEW.path_tokens[0], NEW.id);
```

**Comment by GaryAustin1** [comment]
You don't pass New or any variables into the arguments of a trigger function.   NEW will be available in the function itself.

Also be careful using trigger functions on storage.objects.  The last I checked it did insert to the objects table to test RLS, then stores the file to s3, then does with an update with metadata or deletes the row if the s3 operation fails.  Storage is also changing how they handle things and that might impact how the events work on storage.

**Comment by GaryAustin1** [comment]
The new version of storage has radically changed there trigger methods.
https://github.com/orgs/supabase/discussions/19017#discussioncomment-7613197
Note that all use of triggers on storage schema is at risk of changes...
